File: backend/internal/domain/user_stat.go
package domain

import "fmt"

type UserStat struct {
	Name     string `json:"name"`
	Terminal string `json:"terminal"`
}

func NewHostUserStat(name, terminal string) *UserStat {
	return &UserStat{
		Name:     name,
		Terminal: terminal,
	}
}

func (h UserStat) ToString() string {
	return fmt.Sprintf(
		"Name: %s\nTerminal: %s\n",
		h.Name, h.Terminal)
}

File: frontend/lib/models/user_stat.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'user_stat.freezed.dart';
part 'user_stat.g.dart';

@freezed
class UserStat with _$UserStat {
  const factory UserStat({
    required String name,
    required String terminal,
  }) = _UserStat;

  factory UserStat.fromJson(Map<String, dynamic> json) =>
      _$UserStatFromJson(json);
}
